relay-replays/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+        let payload = include_bytes!("../tests/fixtures/replay_missing_user_ip_address.json");
+        let mut replay_input: ReplayInput = serde_json::from_slice(payload).unwrap();
+        replay_input.set_user_ip_address(ip_address);
+        assert!(*"127.0.0.1" == replay_input.user.ip_address.unwrap());
+    }
+
+    #[test]
+    fn test_set_user_agent_meta() {
+        let payload = include_bytes!("../tests/fixtures/replay.json");
+        let mut replay_input: ReplayInput = serde_json::from_slice(payload).unwrap();
+        replay_input.set_user_agent_meta();
+
+        match replay_input.contexts {
+            Some(contexts) => {
+                match contexts.browser {
+                    Some(browser) => {
+                        assert!(browser.name == *"Safari");
+                        assert!(browser.version.unwrap() == *"15.5");
+                    }
+                    None => unreachable!(),
+                }
+                match contexts.os {
+                    Some(os) => {
+                        assert!(os.name == *"Mac OS X");
+                        assert!(os.version.unwrap() == *"10.15.7");
+                    }
+                    None => unreachable!(),
+                }
+                match contexts.device {
+                    Some(device) => {
+                        assert!(device.family == *"Mac");
+                        assert!(device.brand.unwrap() == *"Apple");
+                        assert!(device.model.unwrap() == *"Mac");
+                    }
+                    None => unreachable!(),
+                }
+            }
+            None => unreachable!(),
+        }
+    }
+
+    #[test]
+    fn test_set_user_agent_meta_no_requests() {
+        let payload = include_bytes!("../tests/fixtures/replay_no_requests.json");
+        let mut replay_input: ReplayInput = serde_json::from_slice(payload).unwrap();
+        replay_input.set_user_agent_meta();
+
+        match replay_input.contexts {
+            Some(contexts) => {
+                match contexts.browser {
+                    Some(browser) => {
+                        assert!(browser.name == *"Other");
+                        assert!(browser.version.is_none());
+                    }
+                    None => unreachable!(),
+                }
+                match contexts.os {
+                    Some(os) => {
+                        assert!(os.name == *"Other");
+                        assert!(os.version.is_none());
+                    }
+                    None => unreachable!(),
+                }
+                match contexts.device {
+                    Some(device) => {
+                        assert!(device.family == *"Other");
+                        assert!(device.brand.is_none());
+                        assert!(device.model.is_none());
+                    }
+                    None => unreachable!(),
+                }
+            }
+            None => unreachable!(),
```

Comment:
nit: You could actually replace all these `unreachable!`s with `unwrap()` like you use on line 249, e.g.
```rust
let contexts = replay_inputs.contexts.unwrap();
```